text/0000-natvis.md

```diff
@@ -376,10 +407,77 @@ developers which may lead to a period of learning the syntax. Since this feature
 would be optional, a consumer of a crate that has natvis definitions for types
 would not need to go through this learning curve. 
 
-Not doing this would keep the existing debugging experience for external Rust crates.
-Most Rust types, outside of the standard library, do not have any debugger views
-defined for them by default which makes them difficult to interpret when viewed
-under a debugger.
+## Alternatives
+
+### Alternative 1: existing -C link-arg flag
+
+Supporting this option would mean that changes to rustc are not necessary.
+The changes would be limited to Cargo, which would be responsible for collecting
+the set of Natvis files and passing `-Clink-arg=/NATVIS:{file-path}` for each
+Natvis file.
+
+The drawbacks for this option is that it will only collect Natvis files for the
+top-most manifest. This will not walk the dependency graph and find all relevant
+Natvis files so this will only work for targets that produce a `DLL` or `EXE` and
+not an `.rlib`.
+
+### Alternative 2: custom build script to set /NATVIS linker flag
+
+Supporting this option would mean that changes to cargo and rustc are not necessary.
+Each individual crate would be able to create a custom build script that would set
+the rustc `link-arg` flag `cargo:rustc-link-arg=/NATVIS:{file-path}` for each Natvis
+file.
+
+The drawbacks for this option is that it would force all Rust developers to manually
+create a build script and ensure it is kept up-to-date whenever the set of Natvis files
+are updated. This also would set the linker argument regardless of the linker being used.
+This could lead to potential build errors if the linker being used returns an error for
```

Comment:
Hmm... don't we already have this situation for all uses of linker flags? I am certain I have seen detection logic to only set msvc flags on the right target

Reply:
This I am not sure about as I do not have much experience around build scripts. I will look into this to see how this can be done.

Reply:
Well, the point is moot due to the other concerns you raised, so maybe just remove the comment about linker flags on the wrong platform?